src/rank_llm/demo/rerank_dataset_with_prebuilt_index.py

```diff
@@ -0,0 +1,23 @@
+import sys
+import os
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+parent = os.path.dirname(SCRIPT_DIR)
+parent = os.path.dirname(parent)
+sys.path.append(parent)
+
+from rank_llm.retrieve.retriever import Retriever
+from rank_llm.retrieve.pyserini_retriever import RetrievalMethod
+
+# By default uses BM25 for retrieval
+dataset_name = "dl19"
+retrieved_results = Retriever.from_dataset_with_prebuit_index(dataset_name)
+print(retrieved_results)
+# TODO: add rerank instead of printing retrieved results
+
+# Users can specify other retrieval methods:
+retrieved_results = Retriever.from_dataset_with_prebuit_index(
```

Comment:
How about `from_dataset_with_pyserini`?

Reply:
We plan on adding custom index dir support, again using Pyserini, wouldn't `from_dataset_with_pyserini` be vague?

Reply:
Yeah, I think I agree with @sahel-sh here, this seems more natural in anticipation for the custom index functionality. Although it is a mouthful.

Reply:
ok then, let's keep it as is for now, we can revisit later.